conans/test/performance/test_large_graph.py

```diff
@@ -0,0 +1,43 @@
+import cProfile
+import json
+import pstats
+import time
+from pstats import SortKey
+
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient
+
+
+def test_large_graph():
+    c = TestClient(cache_folder="T:/mycache")
+    num_test = 40
+    num_pkgs = 40
+
+    """for i in range(num_test):
+        conanfile = GenConanfile(f"test{i}", "0.1")
+        if i > 0:
+            conanfile.with_requires(f"test{i-1}/0.1")
+        c.save({"conanfile.py": conanfile})
+        c.run("create .")
+
+    for i in range(num_pkgs):
+        conanfile = GenConanfile(f"pkg{i}", "0.1").with_test_requires(f"test{num_test-1}/0.1")
+        if i > 0:
+            conanfile.with_requires(f"pkg{i-1}/0.1")
+        c.save({"conanfile.py": conanfile})
+        c.run("create .")
+
+    """
+    t = time.time()
+    pr = cProfile.Profile()
+    pr.enable()
+    c.run(f"install --requires=pkg{num_pkgs - 1}/0.1")
+    pr.disable()
+    print(time.time()-t)
+
+    sortby = SortKey.CUMULATIVE
+    ps = pstats.Stats(pr).sort_stats(sortby)
+    ps.print_stats()
```

Comment:
I like this. I used to do it manually by hand, but this seems so much easier having a test already written!

Reply:
It is still very dirty, because the generation of the test packages takes way longer than the test itself, so I cache it. Definitely not for CI, but having it there to play with it is useful